service/http/response.go

```diff
@@ -59,3 +61,44 @@ func (r *Response) Write(w http.ResponseWriter) error {
 
 	return nil
 }
+
+func handlePushHeaders(h map[string][]string) ([]string, map[string][]string) {
+	var p []string
+	pushHeader, ok := h["http2-push"]
+	if !ok {
+		return p, h
+	}
+
+	for _, v := range pushHeader {
+		p = append(p, v)
+	}
+
+	delete(h, "http2-push")
+
+	return p, h
+}
+
+func handleTrailers(h map[string][]string) map[string][]string {
+	trailers, ok := h["trailer"]
+	if !ok {
+		return h
+	}
+
+	zp := regexp.MustCompile(` *, *`)
+	th := []string{}
+	for _, tr := range trailers {
+		th = append(th, zp.Split(tr, -1)...)
+	}
+
+	delete(h, "trailer")
+
+	for _, n := range th {
+		if v, ok := h[n]; ok {
+			h["Trailer:"+n] = v
```

Comment:
`Trailer:` is a standard Golang prefix for hinting headers to be used as trailers:
>TrailerPrefix is a magic prefix for ResponseWriter.Header map keys that, if present, signals that the map entry is actually for the response trailers, and not the response headers. The prefix is stripped after the ServeHTTP call finishes and the values are sent in the trailers.

from: https://golang.org/pkg/net/http/